specs/checking_account_spec.rb

```diff
@@ -22,59 +22,157 @@
 
   describe "#withdraw" do
     it "Applies a $1 fee each time" do
-      # TODO: Your test code here!
+      initial_deposit = 15
+      withdrawal_amount = 1
+      account = Bank::CheckingAccount.new(1,initial_deposit)
+      initial_balance = account.balance
+      account.withdraw(withdrawal_amount)
+      final_balance = account.balance
+      (initial_balance - final_balance).must_equal withdrawal_amount + 1
     end
 
     it "Doesn't modify the balance if the fee would put it negative" do
-      # TODO: Your test code here!
+      initial_deposit = 15
+      withdrawal_amount = 15
+      account = Bank::CheckingAccount.new(1,initial_deposit)
+      initial_balance = account.balance
+      account.withdraw(withdrawal_amount)
+      final_balance = account.balance
+      (initial_balance - final_balance).must_equal 0
     end
   end
 
   describe "#withdraw_using_check" do
     it "Reduces the balance" do
-      # TODO: Your test code here!
+      account = Bank::CheckingAccount.new(1,30)
+      account.withdraw_using_check(20)
+      account.checks_used.must_equal 1
+      account.balance.must_be :<, 30
     end
 
     it "Returns the modified balance" do
-      # TODO: Your test code here!
+      account = Bank::CheckingAccount.new(1,30)
+      account.withdraw_using_check(20)
```

Comment:
you should probably make this:

`account.withdraw_using_check(20).must_equal 10`

Because the test is that `withdraw_using_check` returns the modified balance.